code/modules/power/singularity/collector.dm

```diff
@@ -40,51 +50,51 @@
-		loaded_tank.air_contents.garbage_collect()
+		return
+	var/gas_drained = min(power_production_drain * drain_ratio * delta_time, loaded_tank.air_contents.gases[/datum/gas/plasma][MOLES])
+	loaded_tank.air_contents.gases[/datum/gas/plasma][MOLES] -= gas_drained
+	loaded_tank.air_contents.assert_gas(/datum/gas/tritium)
+	loaded_tank.air_contents.gases[/datum/gas/tritium][MOLES] += gas_drained
+	loaded_tank.air_contents.garbage_collect()
 
-		var/power_produced = RAD_COLLECTOR_OUTPUT
-		add_avail(power_produced)
-		stored_energy-=power_produced
+	var/power_produced = RAD_COLLECTOR_OUTPUT
+	add_avail(power_produced)
+	stored_energy -= power_produced
 
 /obj/machinery/power/rad_collector/interact(mob/user)
-	if(anchored)
-		if(!src.locked)
-			toggle_power()
-			user.visible_message("<span class='notice'>[user.name] turns the [src.name] [active? "on":"off"].</span>", \
-			"<span class='notice'>You turn the [src.name] [active? "on":"off"].</span>")
-			var/fuel
-			if(loaded_tank)
-				fuel = loaded_tank.air_contents.gases[/datum/gas/plasma]
-			fuel = fuel ? fuel[MOLES] : 0
-			investigate_log("turned [active?"<font color='green'>on</font>":"<font color='red'>off</font>"] by [key_name(user)]. [loaded_tank?"Fuel: [round(fuel/0.29)]%":"<font color='red'>It is empty</font>"].", INVESTIGATE_SINGULO)
-			return
-		else
-			to_chat(user, "<span class='warning'>The controls are locked!</span>")
-			return
+	if(!anchored)
+		return
+	if(locked)
+		to_chat(user, "<span class='warning'>The controls are locked!</span>")
+		return
+	toggle_power()
+	user.visible_message("<span class='notice'>[user.name] turns the [src.name] [active? "on":"off"].</span>", \
+	"<span class='notice'>You turn the [src.name] [active? "on":"off"].</span>")
+	var/fuel
+	if(loaded_tank)
+		fuel = loaded_tank.air_contents.gases[/datum/gas/plasma]
+	fuel = fuel ? fuel[MOLES] : 0
+	investigate_log("turned [active?"<font color='green'>on</font>":"<font color='red'>off</font>"] by [key_name(user)]. [loaded_tank?"Fuel: [round(fuel/0.29)]%":"<font color='red'>It is empty</font>"].", INVESTIGATE_SINGULO)
 
 /obj/machinery/power/rad_collector/can_be_unfasten_wrench(mob/user, silent)
-	if(loaded_tank)
-		if(!silent)
-			to_chat(user, "<span class='warning'>Remove the plasma tank first!</span>")
-		return FAILED_UNFASTEN
-	return ..()
+	if(!loaded_tank)
+		return ..()
+	if(!silent)
+		to_chat(user, "<span class='warning'>Remove the plasma tank first!</span>")
+	return FAILED_UNFASTEN
+
 
 /obj/machinery/power/rad_collector/set_anchored(anchorvalue)
 	. = ..()
 	if(isnull(.))
 		return //no need to process if we didn't change anything.
-	if(anchorvalue)
-		connect_to_network()
-	else
+	if(!anchorvalue)
 		disconnect_from_network()
+		return
+	connect_to_network()
 
-/obj/machinery/power/rad_collector/attackby(obj/item/W, mob/user, params)
-	if(istype(W, /obj/item/tank/internals/plasma))
+/obj/machinery/power/rad_collector/attackby(obj/item/item, mob/user, params)
```

Comment:
in a perfect world we'd replace this for every attackby,  but that's a good way to get tendonitis